Structure-Oncall/OnStaffTeacher.java

```diff
@@ -0,0 +1,19 @@
+public class OnStaffTeacher extends Teacher{
+	int replacementsThisWeek=0;
+	int replacementsThisMonth=0;
+	int totalTally=0;
+	public OnStaffTeacher(String name, Period[] scheldule,String teachable){
+		super(name,scheldule,teachable);
+	}
+	public void incrementWeeklyTally(){
+		replacementsThisWeek++;
+		totalTally++;
+	}
+	public void incrementMonthlyTally(){

```

Comment:
We may want to have just one incrementTally function that increments all of them. That or a separate incrementTotalTally function. If a teacher is assigned and both of these are called then totalTally will be incremented twice.